set.mm

```diff
@@ -674802,6 +674806,554 @@ fixed reference functional determined by this vector (corresponding to
+      ( vx vb vr co wcel cv cfv wceq cn0 wa cvv vk cprimroots cmg c0g cdvds wbr
+      wi wral cbs crab w3a csb ccmn cmpo df-primroots a1i simprl fveq2d simplrl
+      simplrr oveq123d eqeq12d oveqdr breq1d imbi12d ralbidv rabbidva csbeq12dv
+      eqidd anbi12d eqid fvexd rabexd simpr rabeqdv csbied eleq1d mpbird ovmpod
+      eqtrd eleq2d wb oveq2 eqeq1d imbi1d elrab 3anass bicomi biidd oveqd bitrd
+      eqcomi 3anbi123d ) AEBDUBMZNEDJOZBUCPZMZBUDPZQZFOZWOWPMZWRQZDWTUEUFZUGZFR
+      UHZSZJBUIPZUJZNZEXGNZDECMZWRQZWTECMZWRQZXCUGZFRUHZUKZAWNXHEAWNKXGXFJKOZUJ
+      ZULZXHALUABDUMRKLOZUIPZUAOZWOYAUCPZMZYAUDPZQZWTWOYDMZYFQZYCWTUEUFZUGZFRUH
+      ZSZJXRUJZULZXTUBTUBLUAUMRYOUNQAUALJKFUOUPAYABQZYCDQZSSZKYBYNXGXSYRYABUIAY
+      PYQUQZURYRYMXFJXRYRWOXRNZSZYGWSYLXEUUAYEWQYFWRUUAYCDWOWOYDWPUUAYABUCAYPYQ
+      YTUSZURAYPYQYTUTZUUAWOVIVAUUAYABUDUUBURZVBUUAYKXDFRUUAYIXBYJXCUUAYHXAYFWR
+      YRYTFJYDWPYRYABUCYSURVCUUDVBUUAYCDWTUEUUCVDVEVFVJVGVHGHAXTTNXHTNAXFJXGXHT
+      XHVKABUIVLZVMAXTXHTAKXGXSXHTUUEAXRXGQZSXFJXRXGAUUFVNVOVPZVQVRVSUUGVTWAAXI
+      XJDEWPMZWRQZWTEWPMZWRQZXCUGZFRUHZSZSZXQXIUUOWBAXFUUNJEXGWOEQZWSUUIXEUUMUU
+      PWQUUHWRWOEDWPWCWDUUPXDUULFRUUPXBUUKXCUUPXAUUJWRWOEWTWPWCWDWEVFVJWFUPAUUO
+      XJUUIUUMUKZXQUUOUUQWBAUUQUUOXJUUIUUMWGWHUPAXJXJUUIXLUUMXPAXJWIAUUHXKWRAWP
+      CDEWPCQACWPIWLUPZWJWDAUULXOFRAUUKXNXCAUUJXMWRAWPCWTEUURWJWDWEVFWMWKWKWK
+      $.
+  $}
+
+  ${
+    $d A x y $.  $d B x y $.  $d M x y $.  $d ph x y $.
+    mndmolinv.1 $e |- B = ( Base ` M ) $.
+    mndmolinv.2 $e |- ( ph -> M e. Mnd ) $.
+    mndmolinv.3 $e |- ( ph -> A e. B ) $.
+    mndmolinv.4 $e |- ( ph -> E. x e. B ( A ( +g ` M ) x ) = ( 0g ` M ) ) $.
+    $( An element of a monoid that has a right inverse has at most one left
+       inverse.  (Contributed by metakunt, 25-Apr-2025.) $)
+    mndmolinv $p |- ( ph -> E* x e. B ( x ( +g ` M ) A ) = ( 0g ` M ) ) $=
+      ( vy cv cfv co wceq wrex nfv wcel wa syl2anc eqcomd cplusg c0g wral oveq2
+      wi wrmo eqeq1d cbvrexw biimpi syl cmnd ad4antr simplr eqid mndrid simpllr
+      oveq2d eqtrd w3a simp-4r 3jca mndass simpr oveq1d mndlid 3eqtrd ralrimiva
+      ex reximdva mpd rmo2i ) ABKZCEUALZMZEUBLZNZVLJKZNZUEZBDUCZJDOZVPBDUFACVQV
+      MMZVONZJDOZWAACVLVMMZVONZBDOZWDIWGWDWFWCBJDWFJPWCBPVRWEWBVOVLVQCVMUDUGUHU
+      IUJAWCVTJDAVQDQZRZWCVTWIWCRZVSBDWJVLDQZRZVPVRWLVPRZVLVLWBVMMZVNVQVMMZVQWM
+      VLVLVOVMMZWNWMWPVLWMEUKQZWKWPVLNAWQWHWCWKVPGULZWJWKVPUMZDVMEVLVOFVMUNZVOU
+      NZUOSTWMVOWBVLVMWMWBVOWIWCWKVPUPTUQURWMWOWNWMWQWKCDQZWHUSWOWNNWRWMWKXBWHW
+      SAXBWHWCWKVPHULAWHWCWKVPUTZVADVMEVLCVQFWTVBSTWMWOVOVQVMMZVQWMVNVOVQVMWLVP
+      VCVDWMWQWHXDVQNWRXCDVMEVQVOFWTXAVESURVFVHVGVHVIVJVPBJDVPJPVKUJ $.
+  $}
+
+  ${
+    $d R i $.  $d X i $.
+    linvh.1 $e |- ( ph -> X e. ( Base ` R ) ) $.
+    linvh.2 $e |- ( ph -> E! i e. ( Base ` R )
+     ( i ( +g ` R ) X ) = ( 0g ` R ) ) $.
+    $( Slight generalisation for inverses not in a group.  (Contributed by
+       metakunt, 25-Apr-2025.) $)
```

Comment:
I think it's better if the comments for one theorem stand for themselves. Here we don't know what this is a generalization of, without the context.